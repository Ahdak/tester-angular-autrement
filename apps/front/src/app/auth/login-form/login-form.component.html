<mat-card>
  <mat-card-title>Authentification</mat-card-title>
  <mat-card-content>
    <form [formGroup]="form" (ngSubmit)="submit()">
      <p>
        <mat-form-field>
          <input formControlName="username" name="username" type="text" matInput placeholder="Utilisateur" >
          <mat-hint align="start">Client Premium : <strong>user1/pass1</strong></mat-hint>
          <mat-hint align="end">Client normal : <strong>user2/pass2</strong></mat-hint>
        </mat-form-field>
      </p>
      <p>
        <mat-form-field>
          <input formControlName="password" name="password" type="password" matInput placeholder="Mot de passe">
        </mat-form-field>
      </p>
      <p *ngIf="errorMessage" class="loginError" data-testid="errorMessage">
        {{ errorMessage }}
      </p>
      <p class="loginButtons">
        <button [disabled]="!form.valid" type="submit" mat-raised-button>Se connecter</button>
      </p>
    </form>
  </mat-card-content>
</mat-card>
