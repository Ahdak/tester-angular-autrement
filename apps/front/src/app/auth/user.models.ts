export interface AuthenticateBody {
  username: string;
  password: string;
}

export interface AuthenticateResponse {
  accessToken: string;
  user: UserModel;
}

export interface UserModel {
  username: string;
  firstname: string;
  lastname: string;
  isPremium: boolean;
}
