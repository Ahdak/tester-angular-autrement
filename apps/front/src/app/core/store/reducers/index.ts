import { ActionReducer, ActionReducerMap, MetaReducer } from '@ngrx/store';
import * as fromRouter from '@ngrx/router-store';
import { storeFreeze } from 'ngrx-store-freeze';
import { environment } from '../../../../environments/environment';
import { RouterStateUrl } from '../custom-router-state-serializer';
import { localStorageSync } from 'ngrx-store-localstorage';
import { storeLogger } from 'ngrx-store-logger';

export interface State {
  router: fromRouter.RouterReducerState<RouterStateUrl>;
}

export const reducers: ActionReducerMap<State> = {
  router: fromRouter.routerReducer,
};

export function localStorageSyncReducer(
  reducer: ActionReducer<any>,
): ActionReducer<any> {
  return localStorageSync({
    keys: ['auth'],
    rehydrate: true,
  })(reducer);
}

export function logger(reducer: ActionReducer<State>): any {
  return storeLogger()(reducer);
}

export const metaReducers: MetaReducer<State>[] = !environment.production
  ? [localStorageSyncReducer, storeFreeze, logger]
  : [localStorageSyncReducer];
