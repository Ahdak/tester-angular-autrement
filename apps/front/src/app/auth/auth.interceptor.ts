import { Injectable } from '@angular/core';
import {
  HttpInterceptor,
  HttpRequest,
  HttpHandler,
} from '@angular/common/http';
import { mergeMap, take } from 'rxjs/operators';

import { AuthFacade } from './+state/auth.facade';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  static readonly tokenHeaderName = 'x-auth-token';

  constructor(private authFacade: AuthFacade) {}

  intercept(req: HttpRequest<any>, next: HttpHandler) {
    return this.authFacade.accessToken.pipe(
      take(1),
      mergeMap(accessToken => {
        const authReq = req.clone(
          accessToken
            ? {
                headers: req.headers.set(
                  AuthInterceptor.tokenHeaderName,
                  accessToken,
                ),
              }
            : {},
        );
        return next.handle(authReq);
      }),
    );
  }
}
