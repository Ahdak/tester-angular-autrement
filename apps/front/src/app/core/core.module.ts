import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgProgressModule } from '@ngx-progressbar/core';
import { NgProgressRouterModule } from '@ngx-progressbar/router';
import { SharedModule } from '../shared/shared.module';
import { AppStoreModule } from './store/store.module';
import { AuthModule } from '../auth/auth.module';

@NgModule({
  imports: [
    CommonModule,
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    NgProgressModule.forRoot(),
    NgProgressRouterModule,
    AppStoreModule,
    AuthModule,
    SharedModule,
  ],
})
export class CoreModule {}
