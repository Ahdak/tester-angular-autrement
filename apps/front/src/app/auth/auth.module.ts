import { NgModule } from '@angular/core';
import { SharedModule } from '../shared/shared.module';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { authReducers } from './+state/auth.selectors';
import { AuthEffects } from './+state/auth.effects';
import { LoginFormComponent } from './login-form/login-form.component';
import { LoginPageComponent } from './login-page/login-page.component';

@NgModule({
  imports: [
    SharedModule,
    StoreModule.forFeature('auth', authReducers),
    EffectsModule.forFeature([AuthEffects]),
  ],
  declarations: [LoginFormComponent, LoginPageComponent],
})
export class AuthModule {}
