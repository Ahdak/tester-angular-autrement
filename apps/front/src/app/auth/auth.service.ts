import { Injectable } from '@angular/core';
import { Observable, of } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { AuthenticateBody, AuthenticateResponse } from './user.models';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  constructor(private httpClient: HttpClient) {}

  login(body: AuthenticateBody): Observable<AuthenticateResponse> {
    return this.httpClient.post<AuthenticateResponse>(
      '/api/authenticate',
      body,
    );
  }

  logout() {
    return of(true);
  }
}
