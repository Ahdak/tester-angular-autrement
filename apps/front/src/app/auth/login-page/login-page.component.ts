import { Component, OnInit } from '@angular/core';

import { AuthenticateBody } from '../user.models';
import { AuthFacade } from '../+state/auth.facade';

@Component({
  selector: 'app-login-page',
  template: `
    <app-login-form
      (submitted)="onSubmit($event)"
      [pending]="pending$ | async"
      [errorMessage]="error$ | async">
    </app-login-form>
  `,
})
export class LoginPageComponent implements OnInit {
  pending$ = this.authFacade.loginPagePending$;
  error$ = this.authFacade.loginPageError$;

  constructor(private authFacade: AuthFacade) {}

  ngOnInit() {
    this.authFacade.resetLoginPageState();
  }

  onSubmit(body: AuthenticateBody) {
    this.authFacade.login(body);
  }
}
