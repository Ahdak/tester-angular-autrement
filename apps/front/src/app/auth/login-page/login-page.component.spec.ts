import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { combineReducers, StoreModule } from '@ngrx/store';
import { of, throwError } from 'rxjs';

import { LoginPageComponent } from './login-page.component';
import { AuthFacade } from '../+state/auth.facade';
import { authReducers } from '../+state/auth.selectors';
import { LoginFormComponent } from '../login-form/login-form.component';
import { SharedModule } from '../../shared/shared.module';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { EffectsModule } from '@ngrx/effects';
import { AuthEffects } from '../+state/auth.effects';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { Router } from '@angular/router';

describe('TestComponent', () => {
  let component: LoginPageComponent;
  let fixture: ComponentFixture<LoginPageComponent>;
  let nativeElement: Element;

  let httpClient: HttpClient;
  let router: Router;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ LoginPageComponent, LoginFormComponent ],
      imports: [
        StoreModule.forRoot({
          'auth': combineReducers(authReducers)
        }),
        EffectsModule.forRoot([AuthEffects]),
        SharedModule,
        NoopAnimationsModule,
        HttpClientTestingModule,
        RouterTestingModule,
      ],
      providers: [
        AuthFacade,
      ],
      schemas: [NO_ERRORS_SCHEMA]

    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginPageComponent);
    component = fixture.debugElement.componentInstance;
    nativeElement = fixture.nativeElement;

    router = TestBed.get(Router);
    httpClient = TestBed.get(HttpClient);

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match initial snapshot', () => {
    expect(nativeElement).toMatchSnapshot();
  });

  it('should log in and redirect to home page', () => {
    const userResponose = {
      accessToken: 'token',
      user: {
        username: 'user',
        firstname: 'john',
        lastname: 'doe',
        isPremium: true,
      }
    };
    spyOn(httpClient, 'post').and.returnValue(of(userResponose));
    spyOn(router, 'navigate').and.stub();

    const body = {username: 'user', password: 'pass'};
    component.onSubmit(body);

    expect(httpClient.post).toHaveBeenCalledWith('/api/authenticate', body);
    expect(router.navigate).toHaveBeenCalledWith(['/']);

  })

  it('should not log in and display an error for bad credentials', () => {
    spyOn(httpClient, 'post').and.returnValue(throwError(new Error()));
    spyOn(router, 'navigate').and.stub();

    const body = {username: 'user', password: 'wrong'};
    component.onSubmit(body);

    expect(httpClient.post).toHaveBeenCalledWith('/api/authenticate', body);
    expect(router.navigate).not.toHaveBeenCalledWith(['/']);

    fixture.detectChanges();
    expect(nativeElement).toMatchSnapshot(); // Assert Error Message

  })
});
