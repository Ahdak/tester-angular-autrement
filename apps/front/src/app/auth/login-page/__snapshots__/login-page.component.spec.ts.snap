// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should match initial snapshot 1`] = `
<div
  id="root1"
  ng-version="6.1.10"
>
  <app-login-form
    ng-reflect-pending="false"
  >
    <mat-card
      class="mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Authentification
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <form
          class="ng-untouched ng-pristine ng-invalid"
          ng-reflect-form="[object Object]"
          novalidate=""
        >
          <p>
            <mat-form-field
              class="mat-form-field ng-tns-c4-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      aria-describedby="mat-hint-2 mat-hint-3"
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                      formcontrolname="username"
                      id="mat-input-2"
                      matinput=""
                      name="username"
                      ng-reflect-name="username"
                      ng-reflect-placeholder="Utilisateur"
                      ng-reflect-type="text"
                      placeholder="Utilisateur"
                      type="text"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                      <label
                        aria-owns="mat-input-2"
                        class="mat-form-field-label ng-tns-c4-0 mat-empty mat-form-field-empty ng-star-inserted"
                        for="mat-input-2"
                        id="mat-form-field-label-5"
                        ng-reflect-ng-switch="false"
                      >
                        <!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                        <!---->
                        Utilisateur
                        <!--bindings={
  "ng-reflect-ng-switch-case": "true"
}-->
                        <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                      </label>
                    </span>
                  </div>
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                <div
                  class="mat-form-field-underline ng-tns-c4-0 ng-star-inserted"
                >
                  <span
                    class="mat-form-field-ripple"
                  />
                </div>
                <div
                  class="mat-form-field-subscript-wrapper"
                  ng-reflect-ng-switch="hint"
                >
                  <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                  <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
                  <div
                    class="mat-form-field-hint-wrapper ng-tns-c4-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                    style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                  >
                    <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                    <mat-hint
                      class="mat-hint"
                      id="mat-hint-2"
                      ng-reflect-align="start"
                    >
                      Client Premium : 
                      <strong>
                        user1/pass1
                      </strong>
                    </mat-hint>
                    <div
                      class="mat-form-field-hint-spacer"
                    />
                    <mat-hint
                      class="mat-hint mat-right"
                      id="mat-hint-3"
                      ng-reflect-align="end"
                    >
                      Client normal : 
                      <strong>
                        user2/pass2
                      </strong>
                    </mat-hint>
                  </div>
                </div>
              </div>
            </mat-form-field>
          </p>
          <p>
            <mat-form-field
              class="mat-form-field ng-tns-c4-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                      formcontrolname="password"
                      id="mat-input-3"
                      matinput=""
                      name="password"
                      ng-reflect-name="password"
                      ng-reflect-placeholder="Mot de passe"
                      ng-reflect-type="password"
                      placeholder="Mot de passe"
                      type="password"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                      <label
                        aria-owns="mat-input-3"
                        class="mat-form-field-label ng-tns-c4-1 mat-empty mat-form-field-empty ng-star-inserted"
                        for="mat-input-3"
                        id="mat-form-field-label-7"
                        ng-reflect-ng-switch="false"
                      >
                        <!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                        <!---->
                        Mot de passe
                        <!--bindings={
  "ng-reflect-ng-switch-case": "true"
}-->
                        <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                      </label>
                    </span>
                  </div>
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                <div
                  class="mat-form-field-underline ng-tns-c4-1 ng-star-inserted"
                >
                  <span
                    class="mat-form-field-ripple"
                  />
                </div>
                <div
                  class="mat-form-field-subscript-wrapper"
                  ng-reflect-ng-switch="hint"
                >
                  <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                  <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
                  <div
                    class="mat-form-field-hint-wrapper ng-tns-c4-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                    style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                  >
                    <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                    <div
                      class="mat-form-field-hint-spacer"
                    />
                  </div>
                </div>
              </div>
            </mat-form-field>
          </p>
          <!--bindings={
  "ng-reflect-ng-if": null
}-->
          <p
            class="loginButtons"
          >
            <button
              class="mat-raised-button _mat-animation-noopable"
              disabled=""
              mat-raised-button=""
              ng-reflect-disabled="true"
              type="submit"
            >
              <span
                class="mat-button-wrapper"
              >
                Se connecter
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="true"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </p>
        </form>
      </mat-card-content>
    </mat-card>
  </app-login-form>
</div>
`;

exports[`TestComponent should not log in and display an error for bad credentials 1`] = `
<div
  id="root3"
  ng-version="6.1.10"
>
  <app-login-form
    ng-reflect-error-message="Identifiant ou mot de passe in"
    ng-reflect-pending="false"
  >
    <mat-card
      class="mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Authentification
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <form
          class="ng-untouched ng-pristine ng-invalid"
          ng-reflect-form="[object Object]"
          novalidate=""
        >
          <p>
            <mat-form-field
              class="mat-form-field ng-tns-c10-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      aria-describedby="mat-hint-6 mat-hint-7"
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                      formcontrolname="username"
                      id="mat-input-6"
                      matinput=""
                      name="username"
                      ng-reflect-name="username"
                      ng-reflect-placeholder="Utilisateur"
                      ng-reflect-type="text"
                      placeholder="Utilisateur"
                      type="text"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                      <label
                        aria-owns="mat-input-6"
                        class="mat-form-field-label ng-tns-c10-0 mat-empty mat-form-field-empty ng-star-inserted"
                        for="mat-input-6"
                        id="mat-form-field-label-13"
                        ng-reflect-ng-switch="false"
                      >
                        <!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                        <!---->
                        Utilisateur
                        <!--bindings={
  "ng-reflect-ng-switch-case": "true"
}-->
                        <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                      </label>
                    </span>
                  </div>
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                <div
                  class="mat-form-field-underline ng-tns-c10-0 ng-star-inserted"
                >
                  <span
                    class="mat-form-field-ripple"
                  />
                </div>
                <div
                  class="mat-form-field-subscript-wrapper"
                  ng-reflect-ng-switch="hint"
                >
                  <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                  <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
                  <div
                    class="mat-form-field-hint-wrapper ng-tns-c10-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                    style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                  >
                    <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                    <mat-hint
                      class="mat-hint"
                      id="mat-hint-6"
                      ng-reflect-align="start"
                    >
                      Client Premium : 
                      <strong>
                        user1/pass1
                      </strong>
                    </mat-hint>
                    <div
                      class="mat-form-field-hint-spacer"
                    />
                    <mat-hint
                      class="mat-hint mat-right"
                      id="mat-hint-7"
                      ng-reflect-align="end"
                    >
                      Client normal : 
                      <strong>
                        user2/pass2
                      </strong>
                    </mat-hint>
                  </div>
                </div>
              </div>
            </mat-form-field>
          </p>
          <p>
            <mat-form-field
              class="mat-form-field ng-tns-c10-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                      formcontrolname="password"
                      id="mat-input-7"
                      matinput=""
                      name="password"
                      ng-reflect-name="password"
                      ng-reflect-placeholder="Mot de passe"
                      ng-reflect-type="password"
                      placeholder="Mot de passe"
                      type="password"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                      <label
                        aria-owns="mat-input-7"
                        class="mat-form-field-label ng-tns-c10-1 mat-empty mat-form-field-empty ng-star-inserted"
                        for="mat-input-7"
                        id="mat-form-field-label-15"
                        ng-reflect-ng-switch="false"
                      >
                        <!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                        <!---->
                        Mot de passe
                        <!--bindings={
  "ng-reflect-ng-switch-case": "true"
}-->
                        <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                      </label>
                    </span>
                  </div>
                  <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                <div
                  class="mat-form-field-underline ng-tns-c10-1 ng-star-inserted"
                >
                  <span
                    class="mat-form-field-ripple"
                  />
                </div>
                <div
                  class="mat-form-field-subscript-wrapper"
                  ng-reflect-ng-switch="hint"
                >
                  <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                  <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
                  <div
                    class="mat-form-field-hint-wrapper ng-tns-c10-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                    style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                  >
                    <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                    <div
                      class="mat-form-field-hint-spacer"
                    />
                  </div>
                </div>
              </div>
            </mat-form-field>
          </p>
          <!--bindings={
  "ng-reflect-ng-if": "Identifiant ou mot de passe in"
}-->
          <p
            class="loginError ng-star-inserted"
            data-testid="errorMessage"
          >
             Identifiant ou mot de passe incorrect 
          </p>
          <p
            class="loginButtons"
          >
            <button
              class="mat-raised-button _mat-animation-noopable"
              disabled=""
              mat-raised-button=""
              ng-reflect-disabled="true"
              type="submit"
            >
              <span
                class="mat-button-wrapper"
              >
                Se connecter
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="true"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </p>
        </form>
      </mat-card-content>
    </mat-card>
  </app-login-form>
</div>
`;
